rpm.go

```diff
@@ -496,39 +491,108 @@ func (r *RPM) writeFileIndexes(h *index) {
 	h.Add(tagFileLangs, EntryStringSlice(fileLangs))
 }
 
-// AddPretrans adds a pretrans scriptlet
+func (r *RPM) writeScriptlets(h *index) {
+	for name, s := range r.scriptlets {
+		if s.content != "" || s.interpreter != "" {
+			tagInfo, found := scriptletsTypes[name]
+			if !found {
+				panic(fmt.Sprintf("internal error: invalid scriptlet name %q", name))
+			}
+			// If only an interpreter is given then the `scriptlet` becomes a `program` which is called without arguments
+			if s.content != "" {
+				h.Add(tagInfo.tag, EntryString(s.content))
+			}
+			if s.interpreter == "" {
+				interpreter := r.scriptletInterpreter
+				if tagInfo.lua {
+					interpreter = "<lua>"
+				}
+				h.Add(tagInfo.programTag, EntryString(interpreter))
+			} else {
+				h.Add(tagInfo.programTag, EntryString(s.interpreter))
+			}
+
+		}
+	}
+}
+
+// SetScriptletInterpreter sets the scriptlet interpreter (a.k.a. scriptlet program) used to call
+// all scriptlets, defaults to `/bin/sh`. Will reset anything set by `SetScriptletInterpreterFor()`.
+// An emtpy string resets to the default. pretrans and posttrans usually use <lua> and are skipped.
+// The scriptlets of an rpm can be checked via `rpm -qp --scripts RPMFILE`.
+func (r *RPM) SetScriptletInterpreter(interpreter string) {
+	for name, tagInfo := range scriptletsTypes {
+		if tagInfo.lua {
+			continue
+		}
+		r.SetScriptletInterpreterFor(name, interpreter)
+	}
+}
+
+// Set a per-scriptlet interpreter, where `name` must
+// be one of: `prein postin preun postun pretrans posttrans verifyscript`
+func (r *RPM) SetScriptletInterpreterFor(name, interpreter string) error {
+	_, found := scriptletsTypes[name]
+	if !found {
+		return errors.New(fmt.Sprintf("invalid scriptlet name %q", name))
+	}
+	content := ""
```

Comment:
Use the magic of zero values in go to avoid someof this:

```
item := r.scriptlets[name]
item.interpreter = interpreter
r.scriptlets[name] = item
```

Should just work, correctly, no matter if the value existed before. Here, I tested it out for you:
https://go.dev/play/p/ehTutXq-MwD